dts/bindings/pinctrl/nuvoton,npcx-pinctrl.yaml

```diff
@@ -0,0 +1,103 @@
+
+description: |
+  The Nuvoton pin controller is a singleton node responsible for controlling
+  pin function selection and pin properties. The node has the 'pinctrl' node
+  label set in your SoC's devicetree, so you can modify it like this:
+
+    &pinctrl {
+            /* your modifications go here */
+    };
+
+  A group can also specify shared pin properties common to all the specified
+  pins, such as the 'pinmux-locked' property in group 0. Here is a example to
+  lock the pinmux configurations of all selected adc channels.
+
+    &adc0_default {
+      group0 {
+        pinmux = <&alt6_adc0_sl
+                  &alt6_adc2_sl>;
+      };
+      pinmux-locked;
+    };
+
+  Here is a list of supported standard pin properties:
+    - bias-disable: Disable pull-up/down (default behavior, not required).
+    - bias-pull-down: Enable pull-down resistor.
+    - bias-pull-up: Enable pull-up resistor.
+    - drive-push-pull: Output driver is push-pull (default, not required).
+    - drive-open-drain: Output driver is open-drain.
+    - output-high: Set output state high when pin configured.
+    - output-low: Set output state low when pin configured.
+
+  Custom pin properties for npcx series are available also:
+    - pinmux-locked: Lock pinmux configuration for peripheral device
+    - pinmux-gpio: Inverse pinmux back to gpio
+    - low-voltage-input: Select low-voltage input (1p8v) for io and peripheral devices
+
+  An example for NPCX7 family, include the chip level pinctrl DTSI file in the
+  board level DTS:
+
+    #include <nuvoton/npcx/npcx7/npcx7-pinctrl.dtsi>
+
+  We want to use the I2C0_0 port of the NPCX7M6FB controller and enable the
+  internal 3.3V pull-up if its i2c frequency won't exceed 400kHz.
+
+  To change a pin's pinctrl default properties, add a reference to the
+  pin in the board's DTS file and set the properties as below:
+
+    &i2c0_0_default {
+      group0{
+        bias-pull-up; /* Enable internal pull-up for i2c0_0 */
+      };
+    };
+
+    &i2c0_0 {
+      pinctrl-0 = <&i2c0_0_default>;
+      pinctrl-names = "default";
+    }
+
+compatible: "nuvoton,npcx-pinctrl"
+
+include:
+    - name: base.yaml
+    - name: pincfg-node-group.yaml
+      child-binding:
+        child-binding:
+          property-allowlist:
+            - bias-disable
+            - bias-pull-down
+            - bias-pull-up
+            - drive-push-pull
+            - drive-open-drain
+            - output-high
+            - output-low
+
+child-binding:
+    description: NPCX pin controller pin group
+    child-binding:
+      description: |
+        NPCX pin controller pin configuration state nodes
+      properties:
+        pinmux:
+          type: phandles
+          required: false
+          description: Configurations of pinmux selection
+        periph-pupd:
+          type: array
+          required: false
+          description: |
+            A map to PUPD_ENn register/bit that enable pull-up/down of NPCX peripheral devices
```

Comment:
Please add a note that this property should not be changed by the user.

Reply:
Done.